### DISCUSSION TITLE
[WorkAround] Avira Antivirus and issue with "virus" in DLL file of CMS project

### DISCUSSION THREAD

**Discussion by milosh-96** [OrchardCMS/OrchardCore#11864] [General]
**NET Core version: 3.1 LTS**

So I created a solution called "NewsWeb" and then I added OrchardCore CMS web app via "new" dialog in Visual studio. When I set the  name of the VS project to "NewsWeb.Cms" (and later to "NewsWeb.MainWebSite") and tried to build it I got a warning from Avira about HEUR AGEN viruses from DLL files in  obj and bin folders (NewsWeb.Cms.dll and NewsWeb.MainWebSite.dll). 

The workaround (at least in my case) was to just omit dots from the VS project name and it will build fine. The name "NewsWebCms" compiles fine and there are no warnings. I faced this issue on two Win 10 devices and this workaround worked on both devices.

I hope this will save time to someone.

**Comment by hishamco** [comment]
Seems this unrelated to OC, moreover I already have a solution with multiple projects - including dots in the name - compiles fine with .NET 3.0 and above, seems the AntiVirus is sensitive about naming ;)

**Comment by Skrypt** [comment]
This is a HEUR AGEN virus preliminary detection. "Heuristic" means that it is a function of the Avira Antivirus that does try to detect unknown viruses. Of course, you are creating unknown DLLs which can cause this issue. One solution would be to add an exception in your antivirus software for the folders where you create your own DLLs.

There's nothing we can do on our side to prevent this from happening.
Thanks for reporting though as this is good general knowledge advice for people that could encounter the same issue.
